src/main/java/core/basesyntax/NoSuchElementException.java

```diff
@@ -0,0 +1,7 @@
+package core.basesyntax;
+
+public class NoSuchElementException extends RuntimeException {
```

Comment:
You shouldn't create your own exception. It's a custom exception, so you can just import it.